### ISSUE TITLE
Irregular Class Return Values not being Reflected Correctly with Types in TS

### ISSUE THREAD

**Issue by NightshadeX9X** [microsoft/TypeScript#43272]
# Bug Report

Irregular Class Return Values not being Reflected Correctly with Types in TS

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/PQKgBAogHghgtgBwDYFMwEYwmAWAFADGSMAziWABpgDe+Y9YBA9gHYkAuATgK4HtOcAFAEoadBhM4p23TixpgARgC4wAFgBMYAL4BucfW34jeYMEgUAChADCAFQgARVXYAWaAG4wk3NFRKuTNxIACZKaEwAZmDsAJ4IaAAGLCgA7mAiYAC8AHwKKmAs3HCKKJw6ifhmYABKEACCADIAknYAmi7uYF4+fmAAluRRMfFJKemZuTTalXj4+Mxs7GBQ2YVplCL6puYQVrYOzmBunt6+K2ABQaHhYMNxCZQANGCprv0ErpeBwWEoAI7cGDsND8EaPRLUJSqIolMoVKrmOpNVodY5dHrnVaDO7RB5+F5vD5fAFAkHkMGQmbzPBQAB0ijA1QAqgA5PbWexOSA1GoAeRqNNAkFgiFQYC02AWxDIYDaYjwEkWHB4fAEmVoiokDCkMjkYEi3BYfH6rAyok12qtjFYJCYqDpSCYAHNBAAiGzeVBhGDyfpLX0ECLRNpu4Tba06CMMEwmaocg5OTqnXpy77XMKlXHgsYbSZ5QT57pMfohYSzarIlrtZPdM5oeU4+6jMDJPOiKbUalzHvK5axNbjOVbREWTmHWuYtADq6-W7Nx5tQnvT7puek4Ggpg51tFjwlkIVpENatok511MDpt4ltL14rkmAzcU7dU2b4WKZFns-Zcxw8-lBR7GkgA)

### 💻 Code
```ts
class X {
    constructor() {
        return { b: 42 };
    }
}
// EXPECTED: The value X should be of type `new () => { b: number }`
// REALITY: The value X is of type `new () => {}`


const x = new X();
// EXPECTED: The value x should be of type X, which should equate to type `{ b: number }`
// REALITY: The value x is of type X, which equates to `{}`


x.b // UNEXPECTED ERROR
```

### 🙁 Actual behavior

Classes with no properties/methods defined outside the constructor have a return type of `{}`

### 🙂 Expected behavior

I expected classes to be typed correctly, and reflect the type that they actually return.

**Comment by RyanCavanaugh** [comment]
Duplicate #27594
